src/lib.rs

```diff
@@ -2439,6 +2456,38 @@ impl Build {
         Ok(())
     }
 
+    fn add_inherited_rustflags(&self, cmd: &mut Tool, target: &Target) -> Result<(), Error> {
+        let env_os = match self.getenv("CARGO_ENCODED_RUSTFLAGS") {
+            Some(env) => env,
+            // No encoded RUSTFLAGS -> nothing to do
+            None => return Ok(()),
+        };
+
+        let mut cc_flags: Vec<OsString> = env_os
+            .to_string_lossy()
+            .split("\u{1f}")
+            // Strip prefixes from rustc flags
+            .flat_map(|flag| {
+                if flag == "-Z" || flag == "-C" {
+                    None
+                } else if flag.starts_with("-Z") || flag.starts_with("-C") {
+                    Some(&flag[2..])
+                } else {
+                    Some(flag)
+                }
+            })
+            .flat_map(|flag| rustc_to_cc_flag(flag, cmd.family))
+            // Filter out flags not supported by the currently used CC
+            .filter(|flag| {
+                self.is_flag_supported_inner(flag, &cmd.path, &target)
+                    .unwrap_or(false)
+            })
+            .collect();
```

Comment:
It would be nice to have at least a docs link here, it's unclear to me that we're correctly parsing everything?

And if we can't guarantee that we correctly parse everything, it'd be nice to have some reasoning why that is fine? And if it's not fine, then we should probably panic if we found an option that we don't understand?

Reply:
For example, I was worried that this wouldn't catch something like `rustc "-C panic=abort"`, but it seems like, so that's a non-issue since `rustc` doesn't parse it either.

Reply:
Another example that is _actually_ an issue: This doesn't parse `rustc --codegen panic=abort`.

But again, I'm not so worried about that specific case here, that can (and should) be fixed, I'm worried about the correctness / soundness at a high level.

Reply:
> And if we can't guarantee that we correctly parse everything, it'd be nice to have some reasoning why that is fine?

The way I've been thinking about this is that we can pretty much only do best-effort here. There are several compilers here, all with their own changing flags so trying to 100% support everything looks like a losing battle.
At the same time, this is mainly intended to be a convenience feature. Users writing their build scripts in the long run should probably be putting the flags they want themselves explicitly either way.

The important concern is that we don't accidentally break anything. Not passing a flag is always fine since that's the status quo and the user can just do `.flag("foo")` to fix it. If we pass a flag that breaks the build that's obviously an issue so we should avoid that.

That's why I've taken the approach of not panicking and not showing any errors, I see this just as a mechanism for the obvious stuff like "-C lto needs the C code to be built with -flto". This approach also eases the maintenance burden a lot, peoples' build script will not suddenly start crashing due to unexpected or changed flags.